machineconfiguration/v1/types.go

```diff
@@ -0,0 +1,752 @@
+	// controllerCertificates represents the latest available observations of the automatically rotating certificates in the MCO.
+	// +optional
+	ControllerCertificates []ControllerCertificate `json:"controllerCertificates"`
+}
+
+// ControllerCertificate contains info about a specific cert.
+type ControllerCertificate struct {
+	// subject is the cert subject
+	// +kubebuilder:validation:Required
+	Subject string `json:"subject"`
+
+	// signer is the  cert Issuer
+	// +kubebuilder:validation:Required
+	Signer string `json:"signer"`
+
+	// notBefore is the lower boundary for validity
+	// +kubebuilder:validation:Required
+	NotBefore *metav1.Time `json:"notBefore"`
+
+	// notAfter is the upper boundary for validity
+	// +kubebuilder:validation:Required
+	NotAfter *metav1.Time `json:"notAfter"`
+
+	// bundleFile is the larger bundle a cert comes from
+	// +kubebuilder:validation:Required
+	BundleFile string `json:"bundleFile"`
+}
+
+// ControllerConfigStatusCondition contains condition information for ControllerConfigStatus
+type ControllerConfigStatusCondition struct {
+	// type specifies the state of the operator's reconciliation functionality.
+	// +kubebuilder:validation:Required
+	Type ControllerConfigStatusConditionType `json:"type"`
+
+	// status of the condition, one of True, False, Unknown.
+	// +kubebuilder:validation:Required
+	Status corev1.ConditionStatus `json:"status"`
+
+	// lastTransitionTime is the time of the last update to the current status object.
+	// +kubebuilder:validation:Required
+	// +nullable
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// reason is the reason for the condition's last transition.  Reasons are PascalCase
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// message provides additional information about the current condition.
+	// This is only to be consumed by humans.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// ControllerConfigStatusConditionType valid conditions of a ControllerConfigStatus
+type ControllerConfigStatusConditionType string
+
+const (
+	// TemplateControllerRunning means the template controller is currently running.
+	TemplateControllerRunning ControllerConfigStatusConditionType = "TemplateControllerRunning"
+
+	// TemplateControllerCompleted means the template controller has completed reconciliation.
+	TemplateControllerCompleted ControllerConfigStatusConditionType = "TemplateControllerCompleted"
+
+	// TemplateControllerFailing means the template controller is failing.
+	TemplateControllerFailing ControllerConfigStatusConditionType = "TemplateControllerFailing"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ControllerConfigList is a list of ControllerConfig resources
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
```

Comment:
Should be level 1 right?

Reply:
good catch, I got the main type but I missed the List